app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/BrowserFragmentTest.kt

```diff
@@ -19,21 +22,25 @@ import org.junit.runner.RunWith
 @RunWith(AndroidJUnit4::class)
 class BrowserFragmentTest {
 
-    private lateinit var scenario: FragmentScenario<BrowserFragment>
+    private lateinit var activityRule: ActivityScenario<MainActivity>

```

Comment:
does testing the fragment really need the MainActivity in background ? sounds like a dependency that s gonna be annoying in the future if the fragment is not in MainActivity...

Reply:
It was mandatory to make the change with the new clickOnItemChangeFragment test. There might be a better way to do this.

Reply:
yes because the fragment depend on the activity in your case (line 48 of BrowserFragment), this should be removed at some point in the future.
Can you add a TODO there to remember to remove this dependency ?

Reply:
Sure